orhan-homework-week3.js

```diff
@@ -0,0 +1,167 @@
+//1. Part Strings!
+
+let myString = "hello,this,is,a,difficult,to,read,sentence";
+console.log(myString);
+console.log(myString.length);
+console.log(myString.replace(/,/g, " "));
+
+//2. Part Arrays!
+
+let favoriteAnimals = ["blowfish", "capricorn", "giraffe"];
+favoriteAnimals.push("turtle");
+console.log(favoriteAnimals);
+favoriteAnimals.splice(1, 0, "meerkat");
+console.log("The new value of the array will be: " + "blowfish, meerkat, capricorn, giraffe, turtle");
+console.log(favoriteAnimals);
+console.log("The array has a length of: " + favoriteAnimals.length);
+favoriteAnimals.splice(3, 1);
+console.log(favoriteAnimals);
+console.log("The item you are looking for is at index: " + favoriteAnimals.indexOf("meerkat"));
+
+
+//3. Part More JavaScript!
+//1.
+function sum(a, b, c) {
+  return a + b + c;
+}
+
+//2.
+function colorCar(color) {
+  console.log(`a ${color} car`);
+}
+colorCar("red");
+
+//3.
+function makeUser(newObj) {
+  console.log(newObj);
+
+}
+const obj = {
+  name: 'John',
+  age: 24
+};
+makeUser(obj);
+
+//4.
+let vehicleCode = {
+  1: "car",
+  2: "motorbike",
+};
+function vehicleType(color, code) {
+  console.log("a " + color + " " + vehicleCode[code]);
+}
+vehicleType("red", 1);
+
+//5.
+const withoutIf = 3 === 3 ? console.log("yes") : console.log("no");
+
+//6.
+function vehicle(color, code, age) {
+  if (age > 0) {
+    age = "used";
+  } else {
+    age = "new";
+  }
+  console.log("a " + color + " " + age + " " + vehicleCode[code]);
+}
+vehicle("black", 2, 2);
+
+//7.
+let vehicles = ["motorbike", "caravan", "bike"];
+console.log(vehicles);
+//8.
+console.log("Third element of vehicles array is: " + vehicles[2]);
+
+//9.
+function vehicle(color, code, age) {
+  vehicleCode["3"] = "bike";
+  if (age > 1) {
+    age = "used";
+  } else {
+    age = "new";
+  }
+  console.log("a " + color + " " + age + " " + vehicleCode[code]);
+}
+vehicle("green", 3, 1);
+
+//10.
+vehicles.unshift("car");
+let myStr = "Amazing Joe's Garage, we service "
+let vehicles2 = [];
+function myFunc() {
+  for (let i = 0; i < vehicles.length; i++) {
+    vehicles2[i] = (vehicles[i] + "s");
+  }
+  for (i = 0; i < vehicles2.length - 2; i++) {
+    vehicles2[i] += ", ";
+  }
+  for (i = vehicles2.length - 2; i < vehicles2.length - 1; i++) {
+    vehicles2[i] += " and ";
+  }
+  vehicles2[vehicles2.length - 1] += ".";
+  console.log(myStr + vehicles2.join(""));
```

Comment:
A few things here:
1. FYI - `vehicles.unshift("car")'` is an impure function i.e it mutates the array on which it is called. Even though it doesn't have a significant impact in our situation, it is not recommended to use such functions. But this is something you will learn later. 🙂 
2. The first `for` loop has got the correct declaration `let i = 0;` but the other loops don't. Can you explain the reason please?
3. Can we perhaps use only a single loop but have if-else condition inside that loop to append commas, and and the full stop rather than looping thrice? Just an alternative which you can think about

Reply:
Hi @yash-kapila,
I changed my assignment as you recommend